pkg/controller/constraint/constraint_controller.go

```diff
@@ -103,12 +103,17 @@ func (a *Adder) Add(mgr manager.Manager) error {
 	if a.IfWatching != nil {
 		r.ifWatching = a.IfWatching
 	}
+	err = r.reporter.registerCallback(r)
+	if err != nil {
+		return err
+	}
 	return add(mgr, r, a.Events)
 }
 
 type ConstraintsCache struct {
-	mux   sync.RWMutex
-	cache map[string]tags
+	mux           sync.RWMutex
```

Comment:
This file should have no delta except, perhaps, changes to the `ConstraintsCache` struct and its methods.